adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -985,7 +1015,38 @@ private boolean checkPreRequirements(final String resource, final String clientI
 
         return true;
     }
-    
+
+    private boolean checkADFSValidationRequirements(@Nullable final String loginHint)
```

Comment:
@weijjia Introduced this check to verify that if AD FS validation is performed, we have sufficient information and configuration to do so.